crates/iroha_core/src/smartcontracts/isi/nft.rs

```diff
@@ -66,12 +69,15 @@ pub mod isi {
         ) -> Result<(), Error> {
             let nft_id = self.object;
 
+            state_transaction
+                .world
+                .domain(&nft_id.domain)
+                .expect("INTERNAL BUG: Can't find domain of NFT to unregister");
             state_transaction
                 .world
                 .nfts
                 .remove(nft_id.clone())
                 .ok_or_else(|| FindError::Nft(nft_id.clone()))?;
-            state_transaction.world.domain(&nft_id.domain)?;
```

Comment:
Order matters here. If the domain is not found _after_ the NFT is found and removed, that should be considered a bug. But if it happens _before_, it may be due to a bad request from the user rather than a bug